### PR TITLE
Add CI gh-action

### PR DESCRIPTION
**Pull request by SferaDev**
### Summary

Closes: #16 

Tasks:

- Add node version default (same as ``fronted-next``)
- Add codeowners and PR template
- Linter
- Tests
- Compile each module with lerna
- In PRs, publish canary build to NPM
- In main pushes, publish beta build to NPM
- Add a manual release action meanwhile we #17

#### Environment variables

We need to add the environment variables for integration tests in GitHub config. Not sure if we want to have a ``xata`` workspace shared with the team for that? (Useful for debugging)

#### PR checks

Also we need to configure that PRs cannot be merged if the actions fail. This can be done, after merging but we need to remember to do it.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1 @@
+*	@xataio/frontenders
```
Comment (gimenete) [inline .github/CODEOWNERS:1]: This is giving an error in the PR diff. I was thinking that maybe the team is secret but we have this in its configuration

image

🤔

**Review by gimenete — COMMENTED** [review]
Looking good!

**Comment by SferaDev** [comment]
Environment variables pending to be set:

- ``INTEGRATION_TEST_DATABASE_URL``
- ``INTEGRATION_TEST_API_KEY``
- ``INTEGRATION_TEST_BRANCH``
- ``NPM_TOKEN``
- ``GITHUB_TOKEN`` (Should be set automatically)

Haven't tried the comment step (unable locally). It's nothing fancy, if it works we keep it, if it doesn't I'll remove it.

File: .github/CODEOWNERS (hunk 1 above)
Comment (TejasQ) [reply .github/CODEOWNERS:1]: Does the team have write access?

**Review by TejasQ — APPROVED** [review]
Cool, but why Lerna instead of something like Nx or TurboRepo?

**Comment by SferaDev** [comment]
> Cool, but why Lerna instead of something like Nx or TurboRepo?

Fabien had experience with it and it was very simple to set up. We don't really have a complex build system or dependencies right now so most benefits of ``nx`` or ``turborepo`` are not needed (yet).

File: .github/CODEOWNERS (hunk 1 above)
Comment (gimenete) [reply .github/CODEOWNERS:1]: That was it. Thanks!

**Comment by gimenete** [comment]
> We need to add the environment variables for integration tests in GitHub config. Not sure if we want to have a xata workspace shared with the team for that? (Useful for debugging)

Yeah, I think we should do that. Go ahead and create a workspace and try to share with at least the people in the frontenders team.

**Comment by gimenete** [comment]
@SferaDev I've configured the GitHub Actions secrets and I've created the database and its structure here https://app.xata.io/workspaces/sdk-integration-tests-19v8n2/dbs/test/main/tables/teams

**Comment by SferaDev** [comment]
I re-ran an old job and pushed a new commit but it still fails on the integration test.

Notice I named the secrets slightly different than the actual env variables:

```
          XATA_DATABASE_URL: ${{ secrets.INTEGRATION_TEST_DATABASE_URL }}
          XATA_API_KEY: ${{ secrets.INTEGRATION_TEST_API_KEY }}
          XATA_BRANCH: ${{ secrets.INTEGRATION_TEST_BRANCH }}
```

**Comment by gimenete** [comment]
@SferaDev I think it's because `test/integration.test.ts` is using `XATA_DATABASE_BRANCH` and the workflow file is using `XATA_BRANCH`.

For reference, these are the secrets we have 

image

**Review by TejasQ — APPROVED** [review]
🚢 🚢 🚢 🚢 🚢 🚢 🚢 🚢 🚢 🚢 🚢 🚢

**Pull request merged**